docs/source/training.rst

```diff
@@ -10,290 +10,362 @@
-example:
+Then we can instantiate a :class:`~transformers.Trainer` like this:
 
 .. code-block:: python
 
-    from torch.nn import functional as F
-    labels = torch.tensor([1,0])
-    outputs = model(input_ids, attention_mask=attention_mask)
-    loss = F.cross_entropy(outputs.logits, labels)
-    loss.backward()
-    optimizer.step()
+    from transformers import Trainer
 
-Of course, you can train on GPU by calling ``to('cuda')`` on the model and inputs as usual.
+    trainer = Trainer(
+        model=model, args=training_args, train_dataset=small_train_dataset, eval_dataset=small_eval_dataset
+    )
 
-We also provide a few learning rate scheduling tools. With the following, we can set up a scheduler which warms up for
-``num_warmup_steps`` and then linearly decays to 0 by the end of training.
+To fine-tune our model, we just need to call
 
 .. code-block:: python
 
-    from transformers import get_linear_schedule_with_warmup
-    scheduler = get_linear_schedule_with_warmup(optimizer, num_warmup_steps, num_train_steps)
+    trainer.train()
+
+which will start a training that you can follow with a progress bar, which should take a couple of minutes to complete
+(as long as you hav access to a GPU). It won't actually tell you anything useful about how well (or badly) your model
+is performing however as by default, there is no evaluation during training, and we didn't tell the
+:class:`~transformers.Trainer` to compute any metrics. Let's have a look on how to do that now!
 
-Then all we have to do is call ``scheduler.step()`` after ``optimizer.step()``.
+To have the :class:`~transformers.Trainer` compute and report metrics, we need to give it a :obj:`compute_metrics`
+function that takes predictions and labels (grouped in a namedtuple called :class:`~transformers.EvalPrediction`) and
+return a dictionary with string items (the metric names) and float values (the metric values).
+
+The 🤗 Datasets library provides an easy way to get the common metrics used in NLP with the :obj:`load_metric` function.
+here we simply use accuracy. Then we define the :obj:`compute_metrics` function that just convert logits to predictions
+(remember that all 🤗 Transformers models return the logits) and feed them to :obj:`compute` method of this metric.
 
 .. code-block:: python
 
-    loss.backward()
-    optimizer.step()
-    scheduler.step()
+    import numpy as np
+    from datasets import load_metric
+
+    metric = load_metric("accuracy")
 
-We highly recommend using :func:`~transformers.Trainer`, discussed below, which conveniently handles the moving parts
-of training 🤗 Transformers models with features like mixed precision and easy tensorboard logging.
+    def compute_metrics(eval_pred):
+        logits, labels = eval_pred
+        predictions = np.argmax(logits, axis=-1)
+        return metric.compute(predictions=predictions, references=labels)
```

Comment:
I think this function could be a bit better explained.  For example:

- Is it assumed that this callback always takes a single tuple of two elements?  Is there any situation where this could change?
- When does this callback get called?  
- Is there a link to read more about how this works and how this gets called?